include/openspace/navigation/waypoint.h

```diff
@@ -51,10 +52,42 @@ class Waypoint {
 private:
     CameraPose _pose;
     std::string _nodeIdentifier;
-    // to be able to handle nodes with faulty bounding spheres
+    // To be able to handle nodes with faulty bounding spheres
     double _validBoundingSphere = 0.0;
 };
 
+/**
+ * Compute a waypoint from the current camera position.
+ *
+ * \return the computed WayPoint
+ */
+Waypoint waypointFromCamera();
+
+struct NodeCameraStateSpec {
+    std::string identifier;
+    std::optional<glm::dvec3> position;
+    std::optional<double> height;
+    bool useTargetUpDirection = false;
+};
+
+// @TODO (2023-05-16, emmbr) Allow different light sources, not only the 'Sun'
+/**
+ * Compute a waypoint from information about a scene graph node and a previous waypoint,
+ * where the camera will be facing the given node. If there is a 'Sun' node in the scene,
+ * it will possibly be used to compute a position on the lit side of the object.
+ *
+ * \param spec details about the node and state to create the waypoint from. Minimal
+ *             information is the identifier of the node, but a position or height
+ *             above the bounding sphere may also be given.
+ * \param startPoint an optional previous waypoint. If not specified, the current camera
+ *                   position will be used.
+ * \param userLinear if true, the new waypoint will be computed along a straight line
+ *                   from the start waypoint to the scene graph node or position.
+ * \return the computed WayPoint
+ */
+Waypoint computeWaypointFromNodeInfo(const NodeCameraStateSpec& spec,
+    std::optional<const Waypoint> startPoint = std::nullopt, bool useLinear = false);
```

Comment:
-> `std::optional<Waypoint>` would be enough since it is a parameter to the function